### PR TITLE
Add support for AzureAD and Authing OAuth providers.

### PR DESCRIPTION
**Pull request by arvindsvenkat**
Add support for AzureAD and Authing OAuth providers.

### CODE REVIEW

**Comment by arvindsvenkat** [comment]
@calebmshafer , I closed the previous PR and opened this one as it was much easier than fixing the commits. Let me know about the test configuration if you need it.

File: packages/oidc-signin-tool/src/TestUsers.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export interface TestBrowserAuthorizationClientConfiguration {
   redirectUri: string;
   scope: string;
   authority?: string;
+  clientSecret?: string;
```
Comment (calebmshafer) [inline packages/oidc-signin-tool/src/TestUsers.ts:26]: Why is a client secret needed in a OAuth2 Auth Code workflow?

Comment (arvindsvenkat) [reply packages/oidc-signin-tool/src/TestUsers.ts:26]: It's needed for Authing auth code workflows. On an unrelated note, there are some ims oidc clients also that use a client secret for the auth code workflow.

**Review by cjhpaul — APPROVED** [review]

File: packages/oidc-signin-tool/src/TestUsers.ts (hunk 1 above)
Comment (calebmshafer) [reply packages/oidc-signin-tool/src/TestUsers.ts:26]: okay, sounds good.

**Review by calebmshafer — APPROVED** [review]
Approving the changes to get them merged. I'll follow up in a separate PR with the integration test pipeline and some other fixes.

**Pull request merged**